typing_extensions/src_py3/typing_extensions.py

```diff
@@ -982,8 +982,11 @@ def __new__(cls, *args, **kwds):
                 return _generic_new(_next_in_mro(cls), cls, *args, **kwds)
             return _generic_new(cls.__next_in_mro__, cls, *args, **kwds)
 
-    Protocol.__doc__ = Protocol.__doc__.format(bases="Protocol, Generic[T]" if
-                                               OLD_GENERICS else "Protocol[T]")
+    # only update the docstring if docstrings are not being stripped
+    if Protocol.__doc__ is not None:
+        Protocol.__doc__ = Protocol.__doc__.format(
+            bases="Protocol, Generic[T]" if OLD_GENERICS else "Protocol[T]"
+        )
```

Comment:
Can you also add a test for this? (I am not an expert in Travis, but I think it should be possible to run tests with flags).

Also don't forget to sign the CLA.

Reply:
I don't think you can change this in Travis, and pytest seems to not like this (since `-OO` removes `asserts`), but you could change the `exec`s to `exec(compile(..., optimize=2))` which should test things.